testdata/testing.go

```diff
@@ -34,9 +34,13 @@ var benchmarks = []testing.InternalBenchmark{}
 var examples = []testing.InternalExample{}
 
 func main() {
-	m := testing.MainStart(nil, tests, benchmarks, examples)
+	m := testing.MainStart(testdeps{}, tests, benchmarks, examples)
 	exitcode := m.Run()
 	if exitcode != 0 {
 		println("exitcode:", exitcode)
 	}
 }
+
+type testdeps struct{}
+
+func (testdeps) MatchString(pat, str string) (bool, error) { return true, nil }
```

Comment:
That... works, I guess, but could lead to unexpected results.

Reply:
Well, we have a bit of a problem in that I can't actually import `testing/internal/testdeps` here, so I need to fake it.  I figured this was ok since it was just a test package (and I didn't want to pull in all of regexp just for this.)
 The reason I need to do this at all is because otherwise in the `testing` package I was getting a panic for the unchecked type assertion.  I suppose I could fix that to allow `nil` being passed in here, and warn saying that `-run` support is not available in that case.

Reply:
I'll rebase this on dev once #2303 is merged.